### DISCUSSION TITLE
Fan speed control

### DISCUSSION THREAD

**Discussion by khughes5** [omerfaruk-aran/esphome_samsung_hvac_bus#25] [General]
Firstly, Thanks so much @lanwin for putting all the effort into writing this code and its continued development (is there some way to send you a donation for a beer or a coffee or something as a thanks!?). Its my first attempt at esphome, and its been a great introduction! 

Also thanks @bzumik1 for the m5 stack add on.. i had originally built an aliexpress sourced ESP + RS485 module but it turns out the receiving side of the RS485 module was stuffed.. luckily your m5stack solution was published right at this time.. and was so much easier for a beginner!

I can see in the repository that you have been adding fan control support 3 weeks ago. But the example yaml hasn't been updated with reference to it. Is it ready for use? i tried replicating the "Mode" section in my yaml with "FanMode", but that gives me an error when compiling :)

**Comment by bzumik1** [comment]
Hi @khughes5 , first of all thanks for the nice words. The fan speed should be part of climate in HA. It should not be necessary to change the yaml, just recompile it.

![image](https://github.com/lanwin/esphome_samsung_ac/assets/32042186/df6e0191-d2bc-407a-8d7e-23273db08bb1)

**Reply by khughes5** [reply]
Ahh! Thanks.. and sorry for the stupid question.. that should have been obvious. to me. but i was too focused on the onboard web server controls!

**Reply by bzumik1** [reply]
No problem 🙂

**Reply by lanwin** [reply]
@khughes5 thank you for your kind words! I dont think the question is stupid :D 

If you really want to donate something. I think the simples way would be to use paypal.me/lanwin .

**Reply by lanwin** [reply]
Thank you @khughes5!
